pytorch_lightning/utilities/cloud_io.py

```diff
@@ -25,6 +25,9 @@
 
 
 def load(path_or_url: str, map_location=None):
+    if not isinstance(path_or_url, (str, Path)):
+        # any sort of BytesIO or similiar
+        return torch.load(path_or_url, map_location=map_location)
```

Comment:
TIL this was an option. the typehint for `path_or_url` should be updated to reflect this